include/gatekeeper_net.h

```diff
@@ -19,8 +19,19 @@
 #ifndef _GATEKEEPER_NET_H_
 #define _GATEKEEPER_NET_H_
 
+#include <stdint.h>
+
+/* Configuration for the Network. */
+struct net_config {
+	uint16_t	num_rx_queues;
+	uint16_t	num_tx_queues;
+	uint32_t	num_ports;
+};
+
+struct net_config *alloc_and_return_net_conf(void);
+
 int gatekeeper_init_network(void);
 
-void gatekeeper_free_network(void);
+int gatekeeper_free_network(void);
```

Comment:
Ok, this is a good justification for having a global variable. But `gatekeeper_init_network()` should still receive the `struct net_config`. Moreover, the global variable `net_conf` should be the struct itself, not a pointer to it since there will be a single instance of this struct. And `alloc_net_conf()` should just return the address of the global variable.